smlng/RIOT
# Pr/common/build

### Contribution description

This should address my remarks from comment

https://github.com/RIOT-OS/RIOT/pull/10038#pullrequestreview-163752065

* add a .gitignore
* make it mounted by docker
* add a test to run in docker

### Testing procedure

    rm -rf /tmp/riot/build; \
    make -C tests/build_system_builddir_docker clean all test \
        BUILD_DIR=/tmp/riot/build \
        BUILD_IN_DOCKER=1 DOCKER='sudo docker'

It should work without errors.

To make it more visible, the `$(Q)` can be removed in the file creation target.

I put the commit in an order with the test first, so it can be tested without the fix more easily by doing `git checkout` on the test commit. They will need to be re-organized later.

### Issues/PRs references

https://github.com/RIOT-OS/RIOT/pull/10038
